pkg/controller/events.go

```diff
@@ -108,7 +108,7 @@ func (c *Controller) alert(canary *flaggerv1.Canary, message string, metadata bo
 
 		// determine alert provider namespace
 		providerNamespace := canary.GetNamespace()
-		if alert.ProviderRef.Namespace != canary.Namespace {
+		if alert.ProviderRef.Namespace != canary.Namespace && alert.ProviderRef.Namespace != "" {
```

Comment:
Why is this required? If a namespace is not provided for the alert provider ref, `alert.ProviderRef.Namespace` will be blank. Since `canary.Namespace` can never be blank, the current if condition will evaluate to false and the canary namespace will be used to fetch the `AlertProvider`.